packages/itwinui-css/src/stepper/stepper.scss

```diff
@@ -0,0 +1,158 @@
+@import '../style/index';
+@import '../text/index';
+
+@mixin iui-stepper {
+  --iui-stepper-circle-size: #{$iui-l};
+}
+
+@mixin iui-stepper-list {
+  @include iui-reset;
+  display: flex;
+  justify-content: space-around;
+  align-items: flex-start;
+  width: 100%;
+}
+
+@mixin iui-stepper-step {
+  display: flex;
+  flex: 1;
+  align-items: center;
+  flex-direction: column;
+  color: var(--iui-color-foreground-positive);
+  position: relative;
+  counter-increment: section;
+  @include iui-text(small);
+
+  // Track
+  &::before {
+    --_iui-stepper-track-first-half: var(--iui-color-foreground-positive);
+    --_iui-stepper-track-second-half: var(--iui-color-foreground-positive);
+
+    content: '';
+    width: 100%;
+    height: $iui-xxs;
+    position: absolute;
+    top: calc(var(--iui-stepper-circle-size) * 0.5);
+    background: linear-gradient(
+      90deg,
+      var(--_iui-stepper-track-first-half) 50%,
+      var(--_iui-stepper-track-second-half) 50%
+    );
+  }
+
+  // Step
+  &::after {
+    content: counter(section);
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    text-align: center;
+    font-variant-numeric: tabular-nums;
+    width: var(--iui-stepper-circle-size);
+    height: var(--iui-stepper-circle-size);
+    border-radius: $iui-border-radius-round;
+    overflow-wrap: break-word;
+    user-select: none;
+    transition: background-color $iui-speed-fast ease-out, border-color $iui-speed-fast ease-out,
+      color $iui-speed-fast ease-out;
+    border: 1px solid;
+    position: absolute;
+    top: 0;
+    color: inherit;
+  }
+
+  // Completed steps
+  &::after {
+    background-color: var(--iui-color-background-1);
+    border-color: var(--iui-color-foreground-positive);
+    color: var(--iui-color-foreground-positive);
+  }
+
+  // Current step
+  &.iui-current {
+    color: var(--iui-color-foreground-positive);
+    font-weight: $iui-font-weight-semibold;
+
+    &::before {
+      --_iui-stepper-track-first-half: var(--iui-color-foreground-positive);
+      --_iui-stepper-track-second-half: var(--iui-color-background-5);
+    }
+
+    &::after {
+      color: var(--iui-color-foreground-accessory);
+      border-color: var(--iui-color-background-positive);
+      background-color: var(--iui-color-background-positive);
+    }
+  }
+
+  // Future steps
+  &.iui-current ~ & {
+    color: var(--iui-text-color-muted);
+
+    &::before {
+      --_iui-stepper-track-first-half: var(--iui-color-background-5);
+      --_iui-stepper-track-second-half: var(--iui-color-background-5);
+    }
+
+    &::after {
+      border-color: var(--iui-color-background-5);
+      color: inherit;
+    }
+  }
+
+  // First step
+  &:first-of-type::before {
+    --_iui-stepper-track-first-half: transparent;
+  }
+
+  // Last step
+  &:last-of-type::before {
+    --_iui-stepper-track-second-half: transparent !important;
+  }
+
+  // Clickable steps
+  &.iui-clickable {
```

Comment:
I can reduce specificity by making this no longer nested.  However, when trying this, I was getting incorrect results.  So to do this I will need to work on it a bit more.  I'd like to determine if I should continue work on this PR or close it before spending any more time.